### DISCUSSION TITLE
Why does a plugin need a noop index.js?

### DISCUSSION THREAD

**Discussion by laurenskling** [gatsbyjs/gatsby#37344] [Help]
Dear friends at Gatsby.
We are in process of migrating the Strapi source to the Gatsby User Collective.
We are hitting the fact that the plugin fails to resolve without having a noop `index.js` in the folder. It seem to only happen when using themes. I see source-wordpress and such are using noop index files as well.

Does a plugin need the empty index file? And if so, why?

Related issues:
https://github.com/gatsby-uc/plugins/issues/349
https://github.com/gatsby-uc/plugins/pull/345

**Comment by laurenskling** [comment]
A quick followup. If a plugin is inside of the `plugins: []` of a theme, it will need to have an index.js file for it to load correctly. 

Thats means every plugin should always have it, since any theme could load any plugin.

**Comment by KyleAMathews** [comment]
Node requires a package entry file to resolve it. Which is an awkward fit for Gatsby plugins as they're (potentially) a mixture of browser plus node code.

We manually resolve the gatsby-* files internally but node will error if the default index.js entry file isn't there. So adding an empty index.js file is one solution. Another is setting the main field in the package.json to point to the gatsby-node.js file.

This is a pretty good explainer https://bytearcher.com/articles/main-property-in-package.json-defines-entry-point/
